aten/src/ATen/native/Normalization.cpp

```diff
@@ -491,10 +491,27 @@ Tensor batch_norm(
     auto out = input.clone();
     if (weight.defined()) out = out * weight[0];
     if (bias.defined()) out = out + bias[0];
-    return out; 
+    return out;
   }
-  return std::get<0>(at::_batch_norm_impl_index(input, weight, bias, running_mean, running_var,
-                                                training, momentum, eps, cudnn_enabled));
+  auto result = at::_batch_norm_impl_index(
+      input,
+      weight,
+      bias,
+      running_mean,
+      running_var,
+      training,
+      momentum,
+      eps,
+      cudnn_enabled);
+  // Not all implementations of batch norm do inplace modifications of mean and var #33867
```

Comment:
This seems like a pessimization? As in, even if there is an inplace modification, you're always doing an extra `copy_` even if it isn't necessary. I feel like it would be better to fix this at the backend level, whenever appropriate.

Reply:
Nope, if backend does proper in-place `std::get<1>(result)` tensor would be the same as `running_mean` tensor, and copy_ will fast exit.